data_processors/pipeline/services/metadata_srv.py

```diff
@@ -195,3 +196,65 @@ def get_metadata_for_library_runs(library_runs: List[LibraryRun]) -> List[LabMet
     for lib_run in library_runs:
         library_id_list.append(lib_run.library_id)
     return get_metadata_by_keywords_in(libraries=library_id_list)
+
+
+def get_sorted_library_id_by_sequencing_time(library_ids: List[str]) -> str:
+    """
+    Business logic:
+    For given library_id list, you'd like to get which library_id to use for triggering a workflow.
+    Using latest-greatest strategy -- such that
+
+        Check through LibraryRun > SequenceRun(PendingAnalysis)
+        Order library by SequenceRun date_modified descending
+        Return single library_id string; latest by their sequencing time
+
+    Example use case, see https://github.com/umccr/data-portal-apis/issues/547
+
+    :return: str library_id (latest by their sequencing time)  (OR) empty string on any fault condition and upto caller
+    to evaluate on returning empty string
+    """
+    recent_library_id = ""
+
+    lbr_qs: QuerySet = (
+        LibraryRun
+        .objects
+        .filter(library_id__in=library_ids)
```

Comment:
We should also take the `valid_for_analysis` flag into account, but as we are not using it yet, I guess it's OK.

Reply:
Yup, a bit later for this.